src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlStringMethodTranslator.cs

```diff
@@ -243,6 +250,16 @@ public SqlExpression Translate(SqlExpression instance, MethodInfo method, IReadO
             if (method == EndsWith)
                 return TranslateStartsEndsWith(instance, arguments[0], false);
 
+
+            if (method == StringJoinString || method == StringJoinObject || method == StringJoinIEnumerable
```

Comment:
Translating Join overloads which accept arbitrary IEnumerables is too wide. For example, a string is an `IEnumerable<char>`, so this would translate join on string:

```c#
string.Join(',', (IEnumerable<char>)"hello") // returns h,e,l,l,o
```

And this would fail in PG, since text isn't an array on PG.

Another example is a CLR array property which is configured with a value converter, which converts the array value into some arbitrary database type. If we translate that, it would fail.

In other words, this should translate only in cases where the type mapping of the operand is an NpgsqlArrayTypeMapping.